lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -106,6 +193,82 @@ async fn transfer_batches(
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+async fn transfer_snapshot(
+    shard_holder: Arc<LockedShardHolder>,
+    shard_id: ShardId,
+    remote_shard: RemoteShard,
+    snapshots_path: &Path,
+    collection_name: &str,
+    local_rest_address: &Url,
+    temp_dir: &Path,
+    _stopped: Arc<AtomicBool>,
+) -> CollectionResult<()> {
+    let shard_holder_read = shard_holder.read().await;
+
+    // Ensure we have configured a queue proxy
+    let is_queue_proxy = match shard_holder_read.get_shard(&shard_id) {
+        Some(shard_replica_set) => shard_replica_set.is_queue_proxy_local().await,
+        None => false,
+    };
+    if !is_queue_proxy {
+        return Err(CollectionError::service_error(format!(
+            "Shard {shard_id} is not a queue proxy shard, cannot do shard snapshot transfer",
+        )));
+    }
+
+    // Create shard snapshot
+    let snapshot_description = shard_holder_read
+        .create_shard_snapshot(snapshots_path, collection_name, shard_id, temp_dir)
+        .await?;
+
+    // Select local shard snapshot download URL
+    let shard_download_url = local_rest_address
+        .join(&format!(
```

Comment:
let's use Url specialized methods like `set_path` to manipulate with urls

Reply:
Resolved in bf64a6104e63ad9e3a46e4267af4d00fd53edce9. `set_path` does URL encoding where needed. This assumes that the collection and snapshot name don't contain characters like `/`.